apps/user-group-ms/core/src/main/java/it/pagopa/selfcare/user_group/core/UserGroupServiceImpl.java

```diff
@@ -49,12 +51,24 @@ public UserGroupOperations createGroup(UserGroupOperations group) {
         Assert.state(authentication.getPrincipal() instanceof SelfCareUser, "Not SelfCareUser principal");
         Assert.notNull(group, "A group is required");
 
+        checkNameUniqueness(group.getName(), group.getProductId(), group.getInstitutionId());
         UserGroupOperations insert = groupConnector.insert(group);
         log.debug("insert = {}", insert);
         log.trace("createGroup end");
         return insert;
     }
 
+    private void checkNameUniqueness(String groupName, String productId, String institutionId) {
+        UserGroupFilter filter = new UserGroupFilter();
+        filter.setProductId(productId);
+        filter.setInstitutionId(institutionId);
+        filter.setStatus(Arrays.asList(UserGroupStatus.ACTIVE, UserGroupStatus.SUSPENDED));
+
+        Page<UserGroupOperations> existingGroups = groupConnector.findAll(filter, Pageable.unpaged());
+        if (existingGroups.stream().anyMatch(g -> g.getName().equals(groupName))) {
+            throw new ResourceAlreadyExistsException(GROUP_NAME_ALREADY_EXISTS);
+        }
```

Comment:
you could modify the checkNameUniqueness method to exclude the current group's ID when performing the uniqueness check during an update operation. for ex: 

```
Page<UserGroupOperations> existingGroups = groupConnector.findAll(filter, Pageable.unpaged());
    boolean isDuplicate = existingGroups.stream()
        .anyMatch(g -> g.getName().equals(groupName) && !g.getId().equals(currentGroupId));
    if (isDuplicate) {
        log.warn("Attempted to create/update group with duplicate name: {}", groupName);   
        throw new ResourceAlreadyExistsException(GROUP_NAME_ALREADY_EXISTS);
    }
```